### ISSUE TITLE
profiles-config, profiles-run

### ISSUE THREAD

**Issue by melato** [melato/lxops#10]
The lxops config file has two experimental  fields: "profiles-config", "profiles-run".

Their functionality is incomplete and I am not using them.
Either remove them, or complete their implementation.

First document their intended purpose.

**profiles-config** specifies profiles that should be attached to an instance only when configuring it.

For example, when launching an instance, its configuration may include installing software from a disk device that is attached from a profile.  Once the software is installed, its installation files are no longer needed and the profile that attaches them can be removed.

**profiles-run** specified profiles that should be attached to an instance when it is running, but not when it is being configured.
The intent of profiles-run was to be used together with profiles-config to make it possible to have instances with read-only root filesystems.  Suppose that you make a "readonly" copy of the default profile and add "readonly: true" to the root device.
In the lxops config file specify:
```
profiles-config:
- default
profiles-run:
- readonly
```

When launching the instance, the default profile would provide a read-write root filesystem, which can be configured.
Once configuration is complete, the default profile would be removed, and the readonly profile would be added.
You would first have to configure the instance so that it does not attempt to modify its root filesystem while it is running.
To make the root filesystem readonly, you need to reboot the instance, which makes it harder to implement.

I do not have much experience with instances with readonly root filesystems.  It seems like a good idea because it ensures that there are no hidden changes to the instance after it is launched, and it can be safely rebuilt from a new image.  But it is more work to configure such an instance properly and it is more difficult to work with.

I cannot think of any other use cases for profiles-run.

**Comment by melato** [comment]
I am leaning towards removing profiles-run.

To implement profiles-config:
- Check that a profile is not in both profiles and profiles-config.
- "lxops launch" should launch the instance with profiles + profiles-config, configure it, then remove the profiles-config profiles.
- "lxops configure" should add the profiles-config profiles to the instance, configure it, then remove the profiles-config profiles.
- "lxops rebuild" should add the profiles-config profiles to the instance, call "lxc rebuild" or "incus rebuild", configure it, then remove the profiles-config profiles.

**Comment by melato** [comment]
remove both profiles-run and profiles-config, to keep things simple.
If it comes up again, they may be re-implemented properly.
profiles-config is more useful.
There are similar considerations for cloud-config files.
Some cloud-config files could be run when the instance is launched for the first time, but not when it rebuilt.
For now let's keep one set of configuration and let the configuration scripts do the right thing.

**Comment by melato** [comment]
Removed.

**Comment by melato** [comment]
profiles-config would be useful for passing secrets to the container during configuration, and then removing them.
For example, a profile could have a readonly disk device with a .netrc file for authorizing downloads.  After configuration, the profile could be removed, thus removing the secrets.

The workaround is to add and remove such a profile manually, when needed.
